client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -184,9 +188,12 @@ export const deleteMediaSuccess = ( { siteId }, mediaItem ) => ( dispatch ) => {
 	dispatchFluxRemoveMediaItemSuccess( siteId, mediaItem );
 };
 
-export const deleteMediaError = ( { siteId }, error ) => () => {
-	dispatchFluxRemoveMediaItemError( siteId, error );
-};
+export const deleteMediaError = ( { mediaId } ) => [
+	removeNotice( `delete-media-notice-${ mediaId }` ),
```

Comment:
This shouldn't be necessary. Because we have the same `id` set, the previous notice should be automatically removed.

Reply:
See my comment https://github.com/Automattic/wp-calypso/pull/46778#discussion_r512684541